src/plugins/intel_cpu/tests/functional/subgraph_tests/src/matmul_decompress_convert.cpp

```diff
@@ -0,0 +1,193 @@
+#include "transformations/rt_info/decompression.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+using namespace CPUTestUtils;
+using namespace ov::test;
+
+namespace SubgraphTestsDefinitions {
+
+/* This test checks that the ConvertMatMulToFC transformation should work and the MatMul node is converted to the FC node.
+ * The Convert node should be removed on the CPU plugin side.
+
+ * Graph before:
+   ------------             ------------
+   |Input(f32)|             |Input(f16)|
+   ------------             ------------
+        |                        |
+        |         ---------------------------------
+        |         |Convert(decompression f16->f32)|
+        |         ---------------------------------
+        |                        |
+    -----------------------------------------------
+    |                   MatMul                    |
+    -----------------------------------------------
+                          |
+                       --------
+                       |Output|
+                       --------
+
+ * Exec graph:
+   ------------    ------------
+   |Input(f32)|    |Input(f16)|
+   ------------    ------------
+        |               |
+   ----------------------------
+   |      FullyConnected      |
+   ----------------------------
+                 |
+              --------
+              |Output|
+              --------
+*/
+
+using MatMulDecompressConvertParams = std::tuple<
+    std::vector<InputShape>, // input shapes
+    std::pair<bool, bool>    // transposeA, transposeB
+>;
+
+class MatMulDecompressConvertTest : public testing::WithParamInterface<MatMulDecompressConvertParams>,
+                                    virtual public SubgraphBaseTest, public CpuTestWithFusing {
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<MatMulDecompressConvertParams> obj) {
+        std::vector<InputShape> inputShapes;
+        std::pair<bool, bool> transpose;
+
+        std::tie(inputShapes, transpose) = obj.param;
+
+        std::ostringstream result;
+        for (const auto& shape : inputShapes) {
+            result << CommonTestUtils::partialShape2str({shape.first}) << "_";
+        }
+        result << "TS=";
+        for (const auto& shape : inputShapes) {
+            result << "(";
+            if (!shape.second.empty()) {
+                auto itr = shape.second.begin();
+                do {
+                    result << CommonTestUtils::vec2str(*itr);
+                } while (++itr != shape.second.end() && result << "_");
+            }
+            result << ")_";
+        }
+        result << "transpose_a=" << transpose.first << "_";
+        result << "transpose_b=" << transpose.second << "_";
+
+        return result.str();
+    }
+
+protected:
+    template<typename T>
+    void transposeShape(T& shape) {
+        IE_ASSERT(shape.size() > 1);
+        std::swap(*(shape.end() - 1), *(shape.end() - 2));
+    }
+
+    void CheckGraph() const {
```

Comment:
There is `CheckNumberOfNodesWithType` of the same purpose.

Reply:
thanks for the comment, now this method is used in the test